src/pages/Account/AdminVenue.tsx

```diff
@@ -1,62 +1,62 @@
-import React, { useMemo } from "react";
+import React, { useMemo, useCallback } from "react";
 import WithNavigationBar from "components/organisms/WithNavigationBar";
 import { useForm, FieldError } from "react-hook-form";
 import * as Yup from "yup";
-import firebase from "firebase/app";
 import "firebase/functions";
+import { useUser } from "hooks/useUser";
+import { VenueInput, createUrlSafeName, createVenue } from "api/admin";
 
 const LONG_DESCRIPTION_PLACEHOLDER =
   "Describe what is unique and wonderful and sparkling about your venue";
 
-interface FormInputs {
-  name: string;
-  bannerImageUrl: FileList;
-  logoImageUrl: FileList;
-  tagline: string;
-  longDescription: string;
-}
-
-const validationSchema = Yup.object().shape<FormInputs>({
+const validationSchema = Yup.object().shape<VenueInput>({
   name: Yup.string()
     .required("Display name required")
     .test(
       "dfsd",
       "Must have alphanumeric characters",
       (val: string) => val.replace(/\W/g, "").length > 0
     ),
-  bannerImageUrl: Yup.mixed<FileList>()
+  bannerImageFile: Yup.mixed<FileList>()
     .required("Required")
     .test(
       "bannerImageUrl",
       "Image required",
       (val: FileList) => val.length > 0
     ),
-  logoImageUrl: Yup.mixed<FileList>()
+  logoImageFile: Yup.mixed<FileList>()
     .required("Required")
     .test("logoImageUrl", "Image required", (val: FileList) => val.length > 0),
   tagline: Yup.string().required("Required"),
   longDescription: Yup.string().required("Required"),
 });
 
 export const AdminVenue: React.FC = () => {
-  const { register, handleSubmit, errors, watch } = useForm<
-    Partial<FormInputs> // bad typing. If not partial, react-hook-forms should force defaultValues to conform to FormInputs but it doesn't
+  const { register, handleSubmit, errors, watch, formState } = useForm<
+    Partial<VenueInput> // bad typing. If not partial, react-hook-forms should force defaultValues to conform to FormInputs but it doesn't
```

Comment:
Did you try passing a default for everything see if it gets rid of Partial? Not sure it's better but I'm curious

Reply:
yes. It does but it's not enforced by typescript. Also I'm not sure if it's possible to set an input to undefined in the html form but if it it would make defaults futile.